Feature iops 2124 ods extension england odsorganisationrole 26092023v1

johngeo opened the pull request:
Renamed the old feature branch for the ODS changes and have thus had to raise a new pull request.  This has the changes made to the previous pull request <https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/403>.

johngeo committed:
ODS Organisation Role Extension with CodeSystem and ValueSet

johngeo committed:
Update Extension-England-ODSOrganisationRole.xml

Corrected URL from:
https://fhir.nhs.uk/StructureDefinition//England/Extension-England-ODSOrganisationRole

To:

https://fhir.nhs.uk/StructureDefinition//England/Extension-England-ODSOrganisationRole

johngeo committed:
Update Extension-England-ODSOrganisationRole.xml

Another // corrected to /.

johngeo committed:
Merge branch 'stu1_develop' into feature/Feature-ODS-extension-england-odsorganisationrole-26092023v1

# Conflicts:
#	structuredefinitions/Extension-England-ODSOrganisationRole.xml

johngeo committed:
Spacing changes.

johngeo committed:
Correction to URL where "/England/" was in the wrong position.

johngeo committed:
In Extension, removed the status (string) data item.  The ValueSet was renamed as it had a space in the filename.

johngeo committed:
Removed ODS from the FHIR asset name, as per the DA advice (since this is not a clinical concept) it shouldn't be in the filename, plus the ODS programme/service may change name in the future.  Other changes made following review by the NHS E IG team.

digitalhealthmartin commented on codesystems/CodeSystem-England-RoleCode.xml:
```diff
@@ -0,0 +1,24 @@
+<CodeSystem xmlns="http://hl7.org/fhir">
```
Add <?xml version="1.0" encoding="utf-8"?> at the top of the code

digitalhealthmartin commented on structuredefinitions/Extension-England-OrganisationRole.xml:
```diff
@@ -0,0 +1,151 @@
+    </telecom>
+  </contact>
+  <description value="An Extension to describe what an Organisation does and whether this is the primary role of the Organisation." />
+  <purpose value="An Extension to describe what an Organization does and whether this is the primary role of the Organisation." />
```
Organization -> Organisation. This is for consistency with how it's spelt elsewhere

digitalhealthmartin commented on valuesets/ValueSet-England-RoleCode.xml:
```diff
@@ -0,0 +1,28 @@
+<ValueSet xmlns="http://hl7.org/fhir">
```
Add <?xml version="1.0" encoding="utf-8"?> at the top of the code

johngeo replied:
Changes have been made

johngeo committed:
Update Extension-England-OrganisationRole.xml

Organization -> Organisation

KazeemHamzat commented on structuredefinitions/Extension-England-OrganisationRole.xml:
```diff
@@ -0,0 +1,151 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Any need to specify the version of XML? This is not consistent with other files.

johngeo replied:
Hi Kazeem, Martin asked me to add the XML declaration {?xml version="1.0" encoding="utf-8"?}

johngeo committed:
Addd XML declaration: <?xml version="1.0" encoding="UTF-8"?>

digitalhealthmartin commented on codesystems/CodeSystem-England-RoleCode.xml:
I think RoleCode for the CodeSystem and ValueSet is too generic. It would arguably be better if it was OrganisationRoleCode. This is because in the future there could be other types of role codes, for practitioners etc, and these assets may not fit those use cases.

johngeo replied:
OK, I can rename them.

johngeo replied:
Renamed the CodeSystem to include ODS in it's name, title, etc.  The ValueSet can have a generic name, as it may in future reference other CodeSystems.  However, the CodeSystem should be more specific, and it's name should reflect the codes it carries.

digitalhealthmartin reviewed (changes requested):
Question the generic naming of the CodeSystem and ValueSet

johngeo committed:
Renamed the CodeSystem to include ODS in it's name, title, etc...  Changed the descriptions of the VS and CS, and removed their "purpose" information.  Changed the description and purpose of the Extension.

johngeo committed:
Removed <?xml version="1.0" encoding="UTF-8"?> and removed <experimental value="false" /> following NHS E IG standup meeting discussion.

johngeo committed:
Merge pull request #407 from NHSDigital/stu1_develop

Feature-IOPS-2206-Remove Elements: Experimental=false & Purpose and m…

digitalhealthmartin commented on codesystems/CodeSystem-England-ODSOrganisationRoleCode.xml:
Need the file names to be named:

Extension-England-OrganisationRole
ValueSet-England-OrganisationRole
CodeSystem-England-ODSOrganisationRole

johngeo replied:
Hi Martin, I will rename the files.

johngeo committed:
Removed "code" from the filename etc...  So the names of these FHIR assets will be:
Extension-England-OrganisationRole
ValueSet-England-OrganisationRole
CodeSystem-England-ODSOrganisationRole
